src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1397,6 +1401,24 @@ Json::Value CascadiaSettings::ToJson() const
     return json;
 }
 
+std::string_view CascadiaSettings::_getDefaultsJsonHelper()
+{
+    const auto resource = FindResourceW(wil::GetModuleInstanceHandle(), MAKEINTRESOURCEW(IDR_DEFAULTS), RT_RCDATA);
+    const auto loaded = LoadResource(wil::GetModuleInstanceHandle(), resource);
+    const auto sz = SizeofResource(wil::GetModuleInstanceHandle(), resource);
+    const auto ptr = LockResource(loaded);
+    return { reinterpret_cast<const char*>(ptr), sz };
+}
+
+std::string_view CascadiaSettings::_getUserDefaultsJsonHelper()
+{
+    const auto resource = FindResourceW(wil::GetModuleInstanceHandle(), MAKEINTRESOURCEW(IDR_USER_DEFAULTS), RT_RCDATA);
+    const auto loaded = LoadResource(wil::GetModuleInstanceHandle(), resource);
+    const auto sz = SizeofResource(wil::GetModuleInstanceHandle(), resource);
+    const auto ptr = LockResource(loaded);
+    return { reinterpret_cast<const char*>(ptr), sz };
+}

```

Comment:
If we had a `std::string_view _loadStringResource(int id)` function we could replace these 3 copies with just (respectively)
```cpp
_loadStringResource(IDR_DEFAULTS)
_loadStringResource(IDR_USER_DEFAULTS)
_loadStringResource(IDR_ENABLE_COLOR_SELECTION)
```

Reply:
I think it is safe for us to just call it multiple times. We know what it does, and it is free. Not just cheap: free

Reply:
I agree about the helper. We can return a string view even though we don’t own the memory, because it is mapped from the binary.

Reply:
Putting the value of GetModuleHandle in a local will make the code read more easily though… so do consider it! :)